hpx/runtime/serialization/string.hpp

```diff
@@ -34,7 +34,11 @@ namespace hpx { namespace serialization
     void serialize(output_archive & ar, std::basic_string<Char, CharTraits, Allocator> & s, unsigned)
     {
         ar << s.size();
-        if(s.empty()) return;
+        if(s.empty())
+        {
+            HPX_ASSERT(s.size() == 0);
+            return;
+        }
 
         save_binary(ar, s.data(), s.size() * sizeof(Char));
     }
```

Comment:
@K-ballo, you are right about checking the stdlib invariants. save_binary currently assumes a count > 0, that's why we exit early if the string is empty.